### PR TITLE
net-misc/ipsvd: Fix incompatible function pointer types

### PR DESCRIPTION
**Pull request by listout**
Removing conditions for usage of sigsetmask and other such functions and using alternative functions.

Bug: https://bugs.gentoo.org/897946

### CODE REVIEW

File: net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (juippis) [inline net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild:4]: Sorry but this won't do anymore, this is too old. You should update to EAPI 8 while at it this time.

File: net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+HOMEPAGE="http://smarden.org/ipsvd/"
+SRC_URI="http://smarden.org/ipsvd/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="static"
+
+DEPEND=""
+RDEPEND=""
```
Comment (juippis) [inline net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild:18]: These empty variables can be removed.

File: net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+inherit toolchain-funcs flag-o-matic
+
+DESCRIPTION="ipsvd is a set of internet protocol service daemons for Unix"
+HOMEPAGE="http://smarden.org/ipsvd/"
+SRC_URI="http://smarden.org/ipsvd/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="static"
```
Comment (juippis) [inline net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild:15]: EAPI-8 will try to disable static libs, so please check that nothing depends on this package with static libs on. Also, get rid of this use flag :)

File: net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	if use static ; then
+		append-ldflags -static
+	fi
+
+	echo "$(tc-getCC) ${CFLAGS}"  > conf-cc
+	echo "$(tc-getCC) ${LDFLAGS}" > conf-ld
+}
+
+src_compile() {
+	cd "${S}"/src
```
Comment (juippis) [inline net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild:38]:
```suggestion
	cd "${S}"/src || die
```
For error-handling.

Then again I wonder if it's better if you just modify `S` to include `/src`? Although in `src_install` there's some `do*` calls for `../src/` so... :\

**Comment by listout** [comment]
>  so please check that nothing depends on this package with static libs on

How do I do that?

**Comment by juippis** [comment]
There are tools like **equery** from *app-portage/gentoolkit* and **rdep** from *app-portage/mgorny-dev-scripts*, but you can also check excessive https://qa-reports.gentoo.org/ (the section Reverse Dependencies)

https://qa-reports.gentoo.org/output/genrdeps/rindex/net-misc/ipsvd ->
 https://qa-reports.gentoo.org/output/genrdeps/rindex/virtual/inetd

looks fine.

File: net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+src_install() {
+	dobin src/{tcpsvd,udpsvd,ipsvd-cdb}
+	dodoc package/{CHANGES,README}
+
+	doman man/ipsvd-instruct.5 man/ipsvd.7 man/udpsvd.8 \
+		man/tcpsvd.8 man/ipsvd-cdb.8
+
+	insinto html
+	dohtml doc/*.html
```
Comment (juippis) [inline net-misc/ipsvd/ipsvd-1.0.0-r3.ebuild:43]:
```
>>> Source compiled.
>>> Test phase [not enabled]: net-misc/ipsvd-1.0.0-r3

>>> Install net-misc/ipsvd-1.0.0-r3 into /var/tmp/portage/net-misc/ipsvd-1.0.0-r3/image
 * ERROR: net-misc/ipsvd-1.0.0-r3::gentoo failed (install phase):
 *   'dohtml' has been banned for EAPI '8'
 * 
```
:)

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**